### PR TITLE
Release 0.1.0-beta.1

### PR DESCRIPTION
**Pull request by pellared**
Part of https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/471

The is an initial, official beta release,
built on top of [OpenTelemetry .NET](https://github.com/open-telemetry/opentelemetry-dotnet):

- [Core components](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/VERSIONING.md#core-components):
  [`1.2.0`](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.2.0)
- Non-core components: [`1.0.0-rc9.2`](https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/1.0.0-rc9.2)
- `System.Diagnostics.DiagnosticSource`: [`6.0.0`](https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/6.0.0)

### Added

- Support for .NET Framework 4.6.2 and higher.
- Support for .NET Core 3.1.
- Support for .NET 5.0 and 6.0.
- ASP.NET and ASP.NET Core source instrumentations.
- [GraphQL](https://www.nuget.org/packages/GraphQL/) bytecode instrumentation.
- [Microsoft.Data.SqlClient](https://www.nuget.org/packages/Microsoft.Data.SqlClient)
  and [System.Data.SqlClient](https://www.nuget.org/packages/System.Data.SqlClient)
  source instrumentation.
- OTLP, Jaeger, Zipkin and Console trace exporters.
- Global management using environment variables:
  `OTEL_DOTNET_AUTO_HOME`, `OTEL_DOTNET_AUTO_ENABLED`,
  `OTEL_DOTNET_AUTO_INCLUDE_PROCESSES`, `OTEL_DOTNET_AUTO_EXCLUDE_PROCESSES`,
  `OTEL_DOTNET_AUTO_AZURE_APP_SERVICES`.
- Support for OpenTelemetry resource environment variables:
  `OTEL_RESOURCE_ATTRIBUTES`, `OTEL_SERVICE_NAME`.
- Instrumentation management using environment variables:
  `OTEL_DOTNET_AUTO_INTEGRATIONS_FILE`, `OTEL_DOTNET_AUTO_ENABLED_INSTRUMENTATIONS`,
  `OTEL_DOTNET_AUTO_DISABLED_INSTRUMENTATIONS`,
  `OTEL_DOTNET_AUTO_{0}_ENABLED`,
  `OTEL_DOTNET_AUTO_DOMAIN_NEUTRAL_INSTRUMENTATION`,
  `OTEL_DOTNET_AUTO_CLR_DISABLE_OPTIMIZATIONS`,
  `OTEL_DOTNET_AUTO_CLR_ENABLE_INLINING`,
  `OTEL_DOTNET_AUTO_CLR_ENABLE_NGEN`.
- Support for OpenTelemetry exporter related environment variables:
  `OTEL_TRACES_EXPORTER`,
  `OTEL_EXPORTER_OTLP_PROTOCOL`,
- Customization and plugin capabilities which can be configured
  using the following environment variables:
  `OTEL_DOTNET_AUTO_LOAD_AT_STARTUP`,
  `OTEL_DOTNET_AUTO_ADDITIONAL_SOURCES`,
  `OTEL_DOTNET_AUTO_LEGACY_SOURCES`,
  `OTEL_DOTNET_AUTO_INSTRUMENTATION_PLUGINS`.
- `OTEL_DOTNET_AUTO_HTTP2UNENCRYPTEDSUPPORT_ENABLED` environment variable
  which enables `System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport`.
  See the [official Microsoft documentation](https://docs.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-6.0#call-insecure-grpc-services-with-net-core-client)
  for more details.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -142,10 +146,6 @@ See [manual-instrumentation.md](manual-instrumentation.md).
 
 See [troubleshooting.md](troubleshooting.md).
 
-## Contributing
```
Comment (nrcventura) [inline docs/README.md:145]: Should we keep this section in addition to the callout that you moved under "We need you"?

**Review by nrcventura — COMMENTED** [review]
It looks like the submodules got added back in with your last commit. Was that intentional?

**Comment by Kielek** [comment]
Consider rewriting history to fully remove submodules.

**Review by pjanotti — COMMENTED** [review]
LGTM minus the submodule issue. I also like @nrcventura suggestion about the contributing section.

**Comment by pellared** [comment]
> It looks like the submodules got added back in with your last commit. Was that intentional?

Nope. I need to check why I got it... Removed here https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/575/commits/57d1f0fce73426a90ba59c1c2d1a14cbcff81ce0

File: docs/README.md (hunk 1 above)
Comment (pellared) [reply docs/README.md]: Reverted https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/575/commits/3184584c84e984d551f786158e1a8e0b349a7ce6

**Comment by pellared** [comment]
> Consider rewriting history to fully remove submodules.

We are "squash merging" so I do not find it necessary.

**Review by nrcventura — APPROVED** [review]

**Pull request merged**